### PR TITLE
SCALRCORE-31422 - Event Bridge: Provider support

### PR DESCRIPTION
**Pull request by soltysss**
### Changelog
* [ ] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [ ] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

File: scalr/data_source_scalr_event_bridge_integration.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+)
+
+func dataSourceScalrEventBridgeIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description: "This data source is used to retrieve details of a single EventBridge.",
+		ReadContext: dataSourceScalrEventBridgeRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description:  "ID of the EventBridge.",
```
Comment (petroprotsakh) [inline scalr/data_source_scalr_event_bridge_integration.go:21]:
```suggestion
				Description:  "ID of the EventBridge integration.",
```

File: scalr/data_source_scalr_event_bridge_integration.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				AtLeastOneOf: []string{"name"},
+			},
+			"name": {
+				Description:  "Name of the EventBridge.",
```
Comment (petroprotsakh) [inline scalr/data_source_scalr_event_bridge_integration.go:29]:
```suggestion
				Description:  "Name of the EventBridge integration.",
```

File: scalr/data_source_scalr_event_bridge_integration.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scalr/go-scalr"
+)
+
+func dataSourceScalrEventBridgeIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description: "This data source is used to retrieve details of a single EventBridge.",
```
Comment (petroprotsakh) [inline scalr/data_source_scalr_event_bridge_integration.go:16]:
```suggestion
		Description: "This data source is used to retrieve details of a single EventBridge integration.",
```

File: scalr/resource_scalr_event_bridge_integration.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+	if d.HasChange("aws_account_id") {
+		return diag.Errorf("Error updating EventBridge integration %s, AWS account ID cannot be changed.", d.Id())
+	}
+
+	if d.HasChange("region") {
+		return diag.Errorf("Error updating EventBridge integration %s, AWS account ID cannot be changed.", d.Id())
+	}
+
+	return nil
```
Comment (petroprotsakh) [inline scalr/resource_scalr_event_bridge_integration.go:116]: So basically we can't update the EventBridge integration.
All these fields are marked as `ForceNew` anyway, so do we need this function at all? BTW it is not connected as `UpdateContext` function in resource schema.

File: scalr/resource_scalr_event_bridge_integration.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceScalrEventBridgeIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	// Get attributes.чи щось піде не так
```
Comment (petroprotsakh) [inline scalr/resource_scalr_event_bridge_integration.go:62]: 😃

File: scalr/resource_scalr_event_bridge_integration.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func resourceScalrEventBridgeIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	integrationID := d.Id()
+
+	log.Printf("[DEBUG] Read EventBridge integration with ID: %s", integrationID)
+	EventBridgeIntegration, err := scalrClient.EventBridgeIntegrations.Read(ctx, integrationID)
+	if err != nil {
+		log.Printf("[DEBUG] EventBridge integration %s no longer exists", integrationID)
```
Comment (petroprotsakh) [inline scalr/resource_scalr_event_bridge_integration.go:90]: It's better to separate 'not found' from the generic read error, as in https://github.com/Scalr/terraform-provider-scalr/blob/29ef9d1896dcd52e7a083146fb1a3bb4dc76280a/scalr/resource_scalr_tag.go#L48

File: docs/resources/event_bridge_integration.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+title: "scalr_event_bridge_integration"
+categorySlug: "scalr-terraform-provider"
+slug: "provider_resource_scalr_event_bridge_integration"
+parentDocSlug: "provider_resources"
+hidden: false
+order: 25
+---
+## Resource: scalr_event_bridge_integration
+
+Manage the state of EventBrridge integrations in Scalr. Create and destroy.
```
Comment (petroprotsakh) [inline docs/resources/event_bridge_integration.md:11]: Doc is out of sync

File: go.mod [hunk 8]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/hashicorp/terraform-plugin-docs v0.16.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.24.1
 	github.com/hashicorp/terraform-svchost v0.0.0-20200729002733-f050f53b9734
-	github.com/scalr/go-scalr v0.0.0-20240701182247-6041f696e09d
+	github.com/scalr/go-scalr v0.0.0-20240704172455-7d196e31dad8
```
Comment (petroprotsakh) [inline go.mod:9]: @soltysss It's now pointing to an outdated commit

Comment (soltysss) [reply go.mod]: updated

**Review by petroprotsakh — APPROVED** [review]

**Pull request merged**